test: add comments for whatwg-url tests

gautamarora opened the pull request:
Added comments to whatwg-url tests to document that they should not be modified until modifications are made upstream as per guidelines for [Web Platform Tests](https://github.com/nodejs/node/blob/master/doc/guides/writing-tests.md#web-platform-tests)

Fixes: https://github.com/nodejs/node/issues/12793

##### Checklist
- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#commit-message-guidelines)
- [x] _update:_ `make lint`

##### Affected core subsystem(s)
test, url

gibfahn commented on test/parallel/test-whatwg-url-constructor.js:
```diff
@@ -15,7 +15,8 @@ const request = {
 
 /* eslint-disable */
-/* WPT Refs:
+/* The following tests are copied from WPT, modifications to them should be upstreamed first.
```
Looks like this PR only changes 16 files, but looking in `test/parallel/` I see 26 with the `test-whatwg` prefix:

```bash
➜  parallel git:(master) ✗ ❯ ls | grep test-whatwg                                                                                                                             ~/wrk/com/node/test/parallel
test-whatwg-url-constructor.js
test-whatwg-url-domainto.js
test-whatwg-url-historical.js
test-whatwg-url-inspect.js
test-whatwg-url-origin.js
test-whatwg-url-parsing.js
test-whatwg-url-properties.js
test-whatwg-url-searchparams-append.js
test-whatwg-url-searchparams-constructor.js
test-whatwg-url-searchparams-delete.js
test-whatwg-url-searchparams-entries.js
test-whatwg-url-searchparams-foreach.js
test-whatwg-url-searchparams-get.js
test-whatwg-url-searchparams-getall.js
test-whatwg-url-searchparams-has.js
test-whatwg-url-searchparams-inspect.js
test-whatwg-url-searchparams-keys.js
test-whatwg-url-searchparams-set.js
test-whatwg-url-searchparams-sort.js
test-whatwg-url-searchparams-stringifier.js
test-whatwg-url-searchparams-values.js
test-whatwg-url-searchparams.js
test-whatwg-url-setters.js
test-whatwg-url-toascii.js
test-whatwg-url-tojson.js
test-whatwg-url-tostringtag.js
➜  parallel git:(master) ✗ ❯ ls | grep test-whatwg | wc -l                                                                                                                     ~/wrk/com/node/test/parallel
      26
```

refack replied:
I believe the delta are file that do not contain imported code.

gautamarora replied:
thanks for reviewing this @gibfahn. @refack is right, the other files for test-whatwg do not contain imported code, and infact have the comment `Tests below are not from WPT.`

gibfahn commented:
Also thanks for being thorough, but could you remove these lines from the commit message?

```md
##### Checklist
- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] commit message follows [commit
guidelines](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#c
ommit-message-guidelines)

##### Affected core subsystem(s)
test, url
```

gautamarora commented:
@gibfahn - fixed body of commit.
@refack - added changes for text fixtures.

should the documentation for `how to write tests for node.js` also be updated to reflect this new format?

refack commented:
> should the documentation for how to write tests for node.js also be updated to reflect this new format?

That would be a nice to have:
https://github.com/nodejs/node/blob/master/doc/guides/writing-tests.md#web-platform-tests

gautamarora commented:
@refack - updated the guide.

gautamarora commented:
@refack - the body of the commit seems a bit messed up [here](https://github.com/nodejs/node/pull/14355/commits/6c02a0b528eb09d54e11afb1246936c75afe8a1d), i'll try to fix this.

gautamarora commented:
@refack - cleaned up the commit message. Let me know if there is any other clean up is needed.

vsemozhetbyt commented:
CI: https://ci.nodejs.org/job/node-test-pull-request/9237/

TimothyGu commented on doc/guides/writing-tests.md:
````diff
@@ -264,7 +264,8 @@ These imported tests will be wrapped like this:
 ```js
 /* eslint-disable */
-/* WPT Refs:
+/* The following tests are copied from WPT, modifications to them should be upstreamed first.
````
Can you please keep the lines under 80 chars? Not sure why the linter didn't catch that.

gautamarora replied:
I can certainly make the change, the recommendation for that particular comment is as per discussion [here](https://github.com/nodejs/node/pull/12798#pullrequestreview-36256522) cc @joyeecheung for feedback as well.

vsemozhetbyt replied:
Maybe `ignoreComments` in `max-len` is true by default?

gautamarora replied:
nevermind, i could also do something like this:
```
/* eslint-disable */
/* The following tests are copied from WPT, modifications to them should be 
   upstreamed first. Refs:
   https://github.com/w3c/web-platform-tests/blob/54c3502d7b/url/urlsearchparams-constructor.html
   License: http://www.w3.org/Consortium/Legal/2008/04-testsuite-copyright.html
*/
```

making the changes now.

vsemozhetbyt commented:
cc @not-an-aardvark, @silverwind, @trott

Is `ignoreComments` in `max-len` `true` by default?
Is this intended option (omission) in our main `.eslintrc.yaml`?

vsemozhetbyt commented:
Seems `false` [by default](https://github.com/eslint/eslint/blob/729bbcdb1448cb04e166433dfbc9bae52ff08bde/lib/rules/max-len.js#L132).
So linter miss `max-len` > 80 due to:
1. `/* eslint-disable */`
2. `test/fixtures` in the .eslintignore

Should we place `/* eslint-disable */` after comments?

gautamarora commented:
@TimothyGu - pushed a commit to keep lines under 80 chars in tests/fixtures/doc. let me know if this is good now.

not-an-aardvark commented:
> Is `ignoreComments` in `max-len` `true` by default?

No, it's `false` by default (i.e. the rule is strict by default on lines with comments, and allows you to opt out of checking them).

> Is this option (omission) intentional in our main `.eslintrc.yaml`?

¯\\\_(ツ)_/¯

gibfahn commented:
>Should we place `/* eslint-disable */` after comments?

Yes, I think we definitely should.

@gautamarora would you be able to do that?

vsemozhetbyt commented:
@not-an-aardvark Thank you. Sorry for bothering, I've missed we eslintignore `test/fixtures`.

gautamarora commented:
@gibfahn - yes, i will make the changes for this.

gautamarora commented:
@gibfahn @vsemozhetbyt @TimothyGu 

I've moved `eslint: disable` to after the comment in the test files. I also confirmed that once I moved this line to after the comment, the linter did fail my comments for being > 80. I've made the change in the guides as well.

For the test fixtures, the linting is ignored due to `.eslintignore` (as has been noted in [earlier comments](https://github.com/nodejs/node/pull/14355#issuecomment-316242553)), but I have made sure that the new comments are within the 80 char limit.

vsemozhetbyt commented:
CI: https://ci.nodejs.org/job/node-test-pull-request/9242/

Trott commented:
Micro-nit: The first sentence of the comments contains a comma splice:

> The following tests are copied from WPT, modifications to them should be upstreamed first.

The comma should be a period (or perhaps a semicolon):

> The following tests are copied from WPT. Modifications to them should be upstreamed first.

No way this should hold up landing of the PR, though. :-D But if you want to fix it, or if you have to go in and fix other things anyway, then awesome.

gautamarora commented:
@Trott - replaced comma with period :)

refack commented:
@gautamarora I see you are getting the full bikeshed PR treatment 😉. Thank you again for bearing with us...

gautamarora commented:
@refack haha! no worries, i totally get it. Thanks for all the amazing feedback, i learnt much more than just adding comments like linting and running individual tests, so hopefully can contribute more. Let me know if you need anything more from my end.

From what I have understood so far, @vsemozhetbyt will need to start another CI build with the latest changes.

joyeecheung commented:
New CI: https://ci.nodejs.org/job/node-test-pull-request/9254/

refack commented:
> From what I have understood so far, vsemozhetbyt will need to start another CI build with the latest changes.

Anyone of the Collaborators can start a CI, you should ping us when you think the PR is stable.
The next lesson is patience, we keep PRs open at least 48 hours (72 on the weekends) so that everyone would get a chance to comment, as per https://github.com/nodejs/node/blob/master/COLLABORATOR_GUIDE.md#accepting-modifications
Also often change that include documentation received extra scrutiny, since we can't just test them, so human eyes are the only way to make sure they are as correct as possible.

refack commented:
Landed in 9e4ab6c206
@gautamarora You definatly got a warm welcome:
![image](https://user-images.githubusercontent.com/96947/28436926-be227768-6d66-11e7-8e4a-10034ef453c4.png)
and now even GitHub recognizes you as:
![image](https://user-images.githubusercontent.com/96947/28437267-02f629ce-6d68-11e7-8729-3394bd54fa7b.png)
